azurelinuxagent/common/osutil/clearlinux.py

```diff
@@ -79,14 +80,14 @@ def del_root_password(self):
                 passwd_content = fileutil.read_file(passwd_file_path)
                 if not passwd_content:
                     # Empty file is no better than no file
-                    raise FileNotFoundError # pylint: disable=undefined-variable
-            except FileNotFoundError: # pylint: disable=undefined-variable
+                    raise IOError
+            except (IOError, OSError):
```

Comment:
should we be checking that errno is ENOENT?

Reply:
Sorry, I'm not quite sure what this is referring to.

Reply:
I meant the previous code was checking for a specific error (FileNotFoundError); to do the same in the new code we would need to check that IOError.errno == ENOENT

Reply:
That makes sense. I'll make the change.